packages/client/src/components/Actions/ActionsListTable.js

```diff
@@ -0,0 +1,137 @@
+import { useContext } from "react";
+import { useErrorMessage } from "hooks";
+import { formatActionString } from "utils";
+import { useModalContext } from "contexts";
+import { Web3Context } from "contexts/Web3";
+import { SIGNER_RESPONSES } from "constants/enums";
+import Svg from "library/Svg";
+import ApprovalRequired from "./components/ApprovalRequired";
+
+const Row = ({
+  safeData,
+  action,
+  displayIndex,
+  onApprove,
+  onReject,
+  onExecute,
+}) => {
+  const onApproveAction = async (action) => {
+    try {
+      await onApprove(action);
+      closeModal();
+    } catch (error) {
+      showErrorModal(error);
+    }
+  };
+
+  const onRejectAction = async (action) => {
+    try {
+      await onReject(action);
+      closeModal();
+    } catch (error) {
+      showErrorModal(error);
+    }
+  };
+
+  const openApproveOrRejectModal = async (action) => {
+    const view = await getActionView(safeData.address, action.uuid);
+    openModal(
+      <ApprovalRequired
+        safeData={safeData}
+        actionView={view}
+        confirmations={action.signerResponses}
+        onApprove={() => onApproveAction(action)}
+        onReject={() => onRejectAction(action)}
+      />,
+      { headerTitle: "Approval Required" }
+    );
+  };
+  const totalSigned = Object.values(action.signerResponses).filter(
+    (x) => x === SIGNER_RESPONSES.APPROVED
+  ).length;
+  const executionReady = totalSigned >= safeData.threshold;
+  const background = executionReady ? "has-text-danger" : "has-text-warning";
+  const actionPrompt = executionReady
+    ? "Pending Signature"
+    : "Pending Approval";
+  const actionText = executionReady ? "Sign" : "Approve";
+  const actionSvg = executionReady ? "Quill" : "EmptyCheck";
+  const actionFn = executionReady ? onExecute : openApproveOrRejectModal;
+
+  const { getActionView } = useContext(Web3Context);
+  const { showErrorModal } = useErrorMessage();
+  const { openModal, closeModal } = useModalContext();
+
+  return (
+    <>
```

Comment:
Looks like we can remove this empty div and just return tr

Reply:
There are two <tr> tags inside, so it should stay

Reply:
The second row is visible only in mobile view, to display the button, and in the first row the button is visible in desktop view

Reply:
I used Manny's idea from this comment https://github.com/DapperCollectives/VESSEL/pull/118#discussion_r982906870